### PR TITLE
Convert Plane structure to a Vehicle Class

### LINKED ISSUE #33: Convert the `Plane` structure to a class
Currently the Plane is just a structure. This is very constricting for design so we have decided to re-work our entire system to use a Vehicle Class that will be common among any Vehicle that is simulated.

This Vehicle Class will have the following properties:
- A name
- A State Vector for the Vehicle's state data
- A Controls Vector for the Vehicle's control input data
- An array of Models that are affecting the Vehicle (such as an Aerodynamic Model, a Propeller Thrust Model, etc.)
- An enumeration for the State Vector so it is constant across all Vehicles and Simulations

This Vehicle Class will have the following methods:
- A creation method which will need a Name, initial State Vector, number of Control Inputs, and a Model Array
- A method to add Models to the Model Array
- Methods to `get` State Vector parameters or the State Vector itself
- Methods to `get` Controls Vector parameters or the Controls Vector itself
- Methods to `set` State and Controls Vector
- Method to get the Derivatives Vector (same size as State Vector) from the Model Array

### PR DESCRIPTION
**Pull request by TheGreatFez**
Closes #33 

Performed the tasks listed out in the issue by creating a Vehicle class which consists of:
The following properties:
- A name
- A State Vector for the Vehicle's state data
- A Controls Vector for the Vehicle's control input data
- An array of Models that are affecting the Vehicle (such as an Aerodynamic Model, a Propeller Thrust Model, etc.)
- An enumeration for the State Vector so it is constant across all Vehicles and Simulations

The following methods:
- A creation method which will need a Name, initial State Vector, number of Control Inputs, and a Model Array
- A method to add Models to the Model Array
- Methods to `get` State Vector parameters or the State Vector itself
- Methods to `get` Controls Vector parameters or the Controls Vector itself
- Methods to `set` State and Controls Vector
- Method to get the Derivatives Vector (same size as State Vector) from the Model Array

I have also created an additional class that is used as a workaround to enumeration called StateVector_Index which includes the defined index positions of all the values in the state vector that will be used for all vehicles.

### CODE REVIEW

File: Simulation/Classes/StateVector_Index.m [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    V_Z = 6 % Velocity, X
+    V   = [4,5,6] % Velocity, Vector [X,Y,Z]
+    O_P = 7 % Orientation, Pitch
+    O_R = 8 % Orientation, Roll
+    O_Y = 9 % Orientation, Yaw
+    O   = [7,8,9] % Orientation, Vector [Pitch,Roll,Yaw]
+    M   = 10 % Mass
+    B_Cap = 11 % Battery Capacity
+	end
+end
```
Comment (Nufflee) [inline Simulation/Classes/StateVector_Index.m]: As we discussed, we probably want to find a different solution for this. That's out of the scope of this PR so we should probably just open an issue.

Comment (Nufflee) [reply Simulation/Classes/StateVector_Index.m]: Unrelated but there seem to be some indentation issues here. It's a tab vs spaces issue and we should really just stick with one. There seems to be more of this mixing below so you should probably configure your code editor to do this automagically.

File: Simulation/Classes/Vehicle.m [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  methods
+    % Create Vehicle Object
+    %   Inputs:
+    %     Name = Name of Vehicle
+    %     init_stateVector = Initial State of Vehicle
+    %     N_Ctrls = Number of Control Inputs
+    %     modelArray = Array of Models that affect the Vehicle
+    function Vh = Vehicle(Name,init_stateVector,N_Ctrls,modelArray)
+      if (nargin != 4)
```
Comment (Nufflee) [inline Simulation/Classes/Vehicle.m]: Why is this necessary? Doesn't Matlab itself make sure that the correct number of arguments is passed in?

**Review by Nufflee — CHANGES_REQUESTED** [review]
Looks good in general with a few minor changes needed.

File: Simulation/Classes/Vehicle.m (hunk 2 above)
Comment (TheGreatFez) [reply Simulation/Classes/Vehicle.m]: Yes it does, however I wanted this to be more explicit and tell the user what the error is via a custom error message. I found this to be helpful when writing tools at work and thought this may help future users understand the code better.

Let me know your thoughts, I can remove it if you'd like, definitely not necessary.

File: Simulation/Classes/StateVector_Index.m (hunk 1 above)
Comment (TheGreatFez) [reply Simulation/Classes/StateVector_Index.m]: I think I misunderstood one of the settings in the Atom editor settings about how it deals with tabs. I now explicitly tell it to only use spaces for tabs everywhere, hopefully that solves the issue (I don't think Octave has this issue, but we will see).

Comment (TheGreatFez) [reply Simulation/Classes/StateVector_Index.m]: I have removed any tabs and replaced them with spaces on this file

File: Simulation/Classes/StateVector_Index.m [hunk 3]
```diff
@@ -0,0 +1,24 @@
+%% State Vector Indecies
```
Comment (Nufflee) [inline Simulation/Classes/StateVector_Index.m]: I think we've had this before - the red `^M` character, seems to be mixing of CRLF Windows line ending and LF Unix ones, [per this issue](https://github.com/atom/atom/issues/3105).

`Indecies` is also misspelled a few times in this file.

**Review by Nufflee — CHANGES_REQUESTED** [review]

File: Simulation/Classes/StateVector_Index.m (hunk 3 above)
Comment (TheGreatFez) [reply Simulation/Classes/StateVector_Index.m]: Made the file LF explicitly and fixed the spelling

File: Simulation/Classes/Vehicle.m (hunk 2 above)
Comment (TheGreatFez) [reply Simulation/Classes/Vehicle.m]: I have removed the error check since it is redundant. Kept the mass > 0 check since that is not something that Octave checks on its own.

**Review by Nufflee — APPROVED** [review]
